vllm/engine/async_llm_engine.py

```diff
@@ -255,18 +260,26 @@ async def step_async(
                 execute_model_req)
         else:
             output = []
+            
+        # Cache results in engine
+        self.previous_output = output
```

Comment:
There is going be an issue when you rebase over latest main because of my PR that adds caching of python objects into the scheduler. Every scheduler() call, python objects are fetched from the cache and then reused in the next scheduler() call. There are 3 calls inside the scheduler to "self._seq_group_metadata_cache.reset()", "self._scheduler_running_outputs_cache.reset()" and "self._scheduled_seq_group_cache.reset()" that allow the next scheduler() call to reuse the cached objects. In your case it is different, since you can reuse objects after every 2 iterations (and not 1), so you want to call *.reset() every even iteration. I can help with this when you rebase.

Reply:
ok, PR is now rebased. I use cached objects every iteration, it's only skipped in the very first step of vllm server